reframe/core/modules.py

```diff
@@ -109,13 +109,19 @@ def resolve_module(self, name):
         visited = set()
         unvisited = [(name, None)]
         path = []
+        loop_nodes = set()
         while unvisited:
             node, parent = unvisited.pop()
-
             # Adjust the path
             while path and path[-1] != parent:
                 path.pop()
 
+            # Handle modules mappings with self loops
+            if node == parent and node not in loop_nodes:
```

Comment:
Why do you need the `loop_nodes` here?

Reply:
I need it to avoid duplicate modules.

Reply:
Can't you just use an `OrderedSet` for the `ret` value and convert it to a list at return?